.github/workflows/transifex_merge_translations.yml

```diff
@@ -0,0 +1,37 @@
+# Since transifex-synchronization-source is just a staging branch to control
+# updates to Transifex, we need to force merge the translations to master,
+# excluding the English source files
+name: Translations merge to master
+
+on:
+  pull_request:
+    types: [closed]
+    branches:
+      - transifex-synchronization-source
+
+jobs:
+  merge-translations:
+    if: github.event.pull_request.user.login == 'transifex-integration[bot]' && github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout master branch
+        uses: actions/checkout@v4
+        with:
+          ref: master
+
+      - name: Setup Git User
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+
+      - name: Force merge translation directories (excluding English)
+        run: |
+          git fetch --unshallow origin transifex-synchronization-source
+          for dir in language/*/; do
+            if [ -d $dir ] && [ "$(basename $dir)" != "en" ]; then
+              git checkout origin/transifex-synchronization-source -- $dir
+              git add $dir
+            fi
+          done
+          git commit -m "[Transifex] Update translations on master"
+          git push origin master --force
```

Comment:
Do not force push, you can override something that was pushed to master while this action is running. Proper solution would be to on failure resync master and retry using e.g. https://github.com/marketplace/actions/retry-step

Reply:
The language subdirectories should never be updated on master except from this branch, any manual updates would be an exceptional situation and have to be coordinated with the manual merging of translation PRs.

Reply:
You can override whole commits though.

Reply:
Wouldn't the only source of failure be a merge conflict, so instead of force push, use `--ours` somewhere in there?

Reply:
```
$ git push --help
(...)
DESCRIPTION
       Updates remote refs using local refs, while sending objects necessary to complete the given refs.
```

Git push is not doing any merges on remote, there is no way to "merge on remote", you are simply updating to what commit a `master` reference on remote points to.

All merging must be resolved in local checkout, then pushed. Because it **must** to be resolve **locally** (in this case in github actions runner), it's impossible to simply avoid race conditions when doing `git push`, you have to retry.